x-pack/plugins/grokdebugger/public/components/grok_debugger/grok_debugger.js

```diff
@@ -73,6 +73,7 @@ export class GrokDebugger extends React.Component {
   };
 
   simulateGrok = async () => {
+    const notifications = this.props.kibana.services.notifications;
```

Comment:
I'm not familiar with the `withKibana` HOC, but you can also pull `notifications` off `CoreSetup` when you first define the plugin. Here's an example: https://github.com/elastic/kibana/blob/master/x-pack/plugins/index_management/public/plugin.ts#L47

The `notifications` service publishes a `toasts` sub-service which automatically gets configured with `uiSettings`: https://github.com/elastic/kibana/blob/master/src/core/public/notifications/notifications_service.ts#L49

Reply:
Yep, this is doing the same thing, but just accessed in a different way. The `<KibanaContextProvider />` defines what you'd like to provide as `services`, in this case for Grok Debugger it's `coreStart` (`<KibanaContextProvider services={{ ...coreStart }}>`). The HOC, `useKibana` hook etc then all have access to those.

That `index_management` example seems to be doing something very similar. We stuck to the Kibana stuff in metrics / logs as it seemed more "official". [There's documentation here on those](https://github.com/elastic/kibana/tree/master/src/plugins/kibana_react#kibana-react) (if interested).